// cdktf/lib/complex-computed-list.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComplexComputedList = exports.BooleanMap = exports.NumberMap = exports.StringMap = void 0;
const tokens_1 = require("./tokens");
class ComplexComputedAttribute {
    constructor(terraformResource, terraformAttribute) {
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
    }
    /**
     * @experimental
     */
    getStringAttribute(terraformAttribute) {
        return tokens_1.Token.asString(this.interpolationForAttribute(terraformAttribute));
    }
    /**
     * @experimental
     */
    getNumberAttribute(terraformAttribute) {
        return tokens_1.Token.asNumber(this.interpolationForAttribute(terraformAttribute));
    }
    /**
     * @experimental
     */
    getListAttribute(terraformAttribute) {
        return tokens_1.Token.asList(this.interpolationForAttribute(terraformAttribute));
    }
    /**
     * @experimental
     */
    getBooleanAttribute(terraformAttribute) {
        return tokens_1.Token.asString(this.interpolationForAttribute(terraformAttribute));
    }
}
/**
 * @experimental
 */
class StringMap {
    /**
     * @experimental
     */
    constructor(terraformResource, terraformAttribute) {
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
    }
    /**
     * @experimental
     */
    lookup(key) {
        return tokens_1.Token.asString(this.terraformResource.interpolationForAttribute(`${this.terraformAttribute}["${key}"]`));
    }
}
exports.StringMap = StringMap;
/**
 * @experimental
 */
class NumberMap {
    /**
     * @experimental
     */
    constructor(terraformResource, terraformAttribute) {
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
    }
    /**
     * @experimental
     */
    lookup(key) {
        return tokens_1.Token.asNumber(this.terraformResource.interpolationForAttribute(`${this.terraformAttribute}["${key}"]`));
    }
}
exports.NumberMap = NumberMap;
/**
 * @experimental
 */
class BooleanMap {
    /**
     * @experimental
     */
    constructor(terraformResource, terraformAttribute) {
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
    }
    /**
     * @experimental
     */
    lookup(key) {
        return tokens_1.Token.asString(this.terraformResource.interpolationForAttribute(`${this.terraformAttribute}["${key}"]`));
    }
}
exports.BooleanMap = BooleanMap;
/**
 * @experimental
 */
class ComplexComputedList extends ComplexComputedAttribute {
    /**
     * @experimental
     */
    constructor(terraformResource, terraformAttribute, index) {
        super(terraformResource, terraformAttribute);
        this.terraformResource = terraformResource;
        this.terraformAttribute = terraformAttribute;
        this.index = index;
    }
    /**
     * @experimental
     */
    interpolationForAttribute(property) {
        return this.terraformResource.interpolationForAttribute(`${this.terraformAttribute}.${this.index}.${property}`);
    }
}
exports.ComplexComputedList = ComplexComputedList;
