// cdktf/lib/terraform-hcl-module.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TerraformHclModule = void 0;
const terraform_module_1 = require("./terraform-module");
/**
 * @experimental
 */
class TerraformHclModule extends terraform_module_1.TerraformModule {
    /**
     * @experimental
     */
    constructor(scope, id, options) {
        super(scope, id, options);
        this._variables = options.variables;
    }
    /**
     * @experimental
     */
    get variables() {
        return this._variables;
    }
    /**
     * @experimental
     */
    set(variable, value) {
        if (!this._variables) {
            this._variables = {};
        }
        this._variables[variable] = value;
    }
    /**
     * @experimental
     */
    get(output) {
        return this.interpolationForOutput(output);
    }
    /**
     * @experimental
     */
    getString(output) {
        return this.get(output);
    }
    /**
     * @experimental
     */
    getNumber(output) {
        return this.get(output);
    }
    /**
     * @experimental
     */
    getBoolean(output) {
        return this.get(output);
    }
    /**
     * @experimental
     */
    getList(output) {
        return this.get(output);
    }
    /**
     * @experimental
     */
    synthesizeAttributes() {
        return { ...this.variables };
    }
}
exports.TerraformHclModule = TerraformHclModule;
